zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbDeploymentState.java

```diff
@@ -151,4 +151,17 @@ public void foreachPendingDeploymentDistribution(
           pendingDeploymentVisitor.visit(deploymentKey, partitionId, lastDeployment.get());
         });
   }
+
+  @Override
+  public DeploymentRecord nextDeployment(final long previousDeploymentKey) {
+    final var nextRawDeployment = new MutableReference<DeploymentRaw>();
+    deploymentKey.wrapLong(previousDeploymentKey + 1);
+    deploymentRawColumnFamily.whileTrue(
+        deploymentKey,
+        (deploymentKey, rawDeployment) -> {
+          nextRawDeployment.set(rawDeployment);
+          return false;
+        });
+    return nextRawDeployment.get() == null ? null : nextRawDeployment.get().getDeploymentRecord();
```

Comment:
Can you elaborate why that's needed?

Reply:
Sure. The `deploymentRawColumnFamily` with a `DeploymentRaw` object. The reference to this object is maintained. When we fetch any data from this CF it will modify this object. That means if you were to do this:

```java
    deploymentState.storeDeploymentRecord(1, deployment1);
    deploymentState.storeDeploymentRecord(2, deployment2);
    deploymentState.storeDeploymentRecord(3, deployment2);

    final var nextDeployment = deploymentState.nextDeployment(1);
    final var nextNextDeployment = deploymentState.nextDeployment(2);
```

`nextDeployment` references the same object as `nextNextDeployment`. If you try and read any data from it you will actually be reading the data from `nextNextDeployment`. 

Is it a problem with the current code? No, because you never call this method twice. However, who knows what will happen in the future. If we start using this it could cause some hard to root-cause bugs. Therefore, copying the data is always the better option.

Reply:
> Therefore, copying the data is always the better option.

Except for performance... I don't think we should always copy everything, otherwise our whole Db/ColumnFamily abstraction doesn't make sense because it could always return copies instead of modifying a reused object

Reply:
We only call this method when scaling up which is something that doesn't happen often. I'd prefer safety over the small performance hit.

Reply:
Addressed in b876679a56d2b4caf7fc3eb2cf68c893776655b5